lib/kitchen/driver/vra.rb

```diff
@@ -60,11 +61,16 @@ def create(state)
 
         server = request_server
         state[:resource_id] = server.id
-
-        ip_address = server.ip_addresses.first
-        raise 'No IP address returned for the vRA request' if ip_address.nil?
-        state[:hostname] = server.ip_addresses.first
-        state[:ssh_key]  = config[:private_key_path] unless config[:private_key_path].nil?
+        if config[:use_dns]
+          dns_name = server.name
```

Comment:
Since you use the `dns_name` variable here, we should use it in line 67 as well, or perhaps don't set an extra variable.

Option 1:
```ruby
raise 'No server name returned for the vRA request' if server.name.nil?
state[:hostname] = server.name
```

or Option 2:
```ruby
dns_name = server.name
raise 'No server name returned for the vRA request' if dns_name.nil?
state[:hostname] = dns_name
```